Hazel/src/Platform/OpenGL/OpenGLVertexArray.cpp

```diff
@@ -26,7 +26,7 @@ namespace Hazel {
 		return 0;
 	}
 
-	OpenGLVertexArray::OpenGLVertexArray()
+	OpenGLVertexArray::OpenGLVertexArray() : m_IndexOffset(0)
```

Comment:
Also as general code review; I feel like this should become:
```cpp
OpenGLVertexArray::OpenGLVertexArray()
    : m_IndexOffset(0)
{
    glCreateVertexArrays(1, &m_RendererID);
}
```
mainly because Cherno prefers doing this and it would be best to stay consistent.

While this probably isn't needed it still would be nice to stay consistent with other code.

Reply:
I'm debating myself if this is even necessary at all, as it doesn't require constructor arguments to initialize that variable, just setting =0 in the header file should have the same effect?

Reply:
That is true, either one of them would work but I think it would be best to set it to 0 within the header file instead. @kcourtney77 could've done this for a reason though, so clarification would be nice. I'm pretty sure they did it based on preference though.

Reply:
Yeah I do that just as a preference. I could definitely change it if you think I should.

Reply:
If you leave it as an initializer list, for sure put it on a new line to be inline with the coding style (pun not intended). I've seen cherno do both header and initialiser list, so that's totaly up to you

Reply:
I think you should put it on a new line like my original post states but since there is no performance difference with declaring it directly on the header file (doing: `` = 0`` like Lovely stated) or doing ``: m_IndexOffset(0)`` either one of them would work. If you want to stay perfectly consistent with Cherno's style it would probably be better to do it via header but it wouldn't matter at all; unless Lovely thinks it would matter (as he is a collaborator) then you probably can keep it the same and just doing what I said.

**tl;dr** It's fine for you to keep it like that (unless Lovely says otherwise) but you should put it on a new line like my original review states.

Reply:
I was about to find an example of both cases, but for a constant value, I only found it defined in the header file. In all other cases I find initializer lists always depending on parameters passed to the constructor...

https://github.com/TheCherno/Hazel/blob/f52b813db65cd7275da56003c501914ad1c7515a/Hazel/src/Hazel/LayerStack.h#L25

If I would go back to the argument about the header file making stuff clear, I'm more thinking about putting it in the header... But whatever you choose I'll accept, as it doesn't impact performance in any way...